pkg/collector/corechecks/oracle-dbm/oracle_dictionary_test.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build oracle_test
+
+package oracle
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	_ "github.com/godror/godror"
+)
+
+var chk Check
+
+var HOST = "localhost"
+var PORT = 1521
+var USER = "c##datadog"
+var PASSWORD = "datadog"
+var SERVICE_NAME = "XE"
+var TNS_ALIAS = "XE"
+var TNS_ADMIN = "/Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/testutil/etc/netadmin"
+
+func TestBasic(t *testing.T) {
+	chk = Check{}
+
+	// language=yaml
+	rawInstanceConfig := []byte(fmt.Sprintf(`
+server: %s
+port: %d
+username: %s
+password: %s
+service_name: %s
+tns_alias: %s
+tns_admin: %s
+dbm: true
+`, HOST, PORT, USER, PASSWORD, SERVICE_NAME, TNS_ALIAS, TNS_ADMIN))
+
+	err := chk.Configure(aggregator.GetSenderManager(), integration.FakeConfigHash, rawInstanceConfig, []byte(``), "oracle_test")
+	require.NoError(t, err)
+
+	assert.Equal(t, chk.config.InstanceConfig.Server, HOST)
+	assert.Equal(t, chk.config.InstanceConfig.Port, PORT)
+	assert.Equal(t, chk.config.InstanceConfig.Username, USER)
+	assert.Equal(t, chk.config.InstanceConfig.Password, PASSWORD)
+	assert.Equal(t, chk.config.InstanceConfig.ServiceName, SERVICE_NAME)
+	assert.Equal(t, chk.config.InstanceConfig.TnsAlias, TNS_ALIAS)
+	assert.Equal(t, chk.config.InstanceConfig.TnsAdmin, TNS_ADMIN)
```

Comment:
`chk.config.InstanceConfig` contains lots of defaults, such as `ObfuscatorOptions:obfuscate.SQLConfig`, that are causing an assertion error, because those defaults aren't initialized in `config.InstanceConfig`:

```
=== FAIL: pkg/collector/corechecks/oracle-dbm TestBasic (0.00s)
    oracle_dictionary_test.go:52: 
        	Error Trace:	/Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/oracle_dictionary_test.go:52
        	Error:      	Not equal: 
        	            	expected: config.InstanceConfig{Server:"localhost", Port:1521, ServiceName:"XE", Username:"c##datadog", Password:"datadog", TnsAlias:"XE", TnsAdmin:"/Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/testutil/etc/netadmin", Protocol:"", Wallet:"", DBM:true, Tags:[]string(nil), LogUnobfuscatedQueries:false, ObfuscatorOptions:obfuscate.SQLConfig{DBMS:"", TableNames:false, CollectCommands:false, CollectComments:false, ReplaceDigits:false, KeepSQLAlias:false, DollarQuotedFunc:false, Cache:false}, InstantClient:false, ReportedHostname:"", QuerySamples:config.QuerySamplesConfig{Enabled:false, IncludeAllSessions:false}, QueryMetrics:config.QueryMetricsConfig{Enabled:false, CollectionInterval:0, DBRowsLimit:0, PlanCacheRetention:0, DisableLastActive:false}, SysMetrics:config.SysMetricsConfig{Enabled:false}, Tablespaces:config.TablespacesConfig{Enabled:false}, ProcessMemory:config.ProcessMemoryConfig{Enabled:false}, SharedMemory:config.SharedMemoryConfig{Enabled:false}, ExecutionPlans:config.ExecutionPlansConfig{Enabled:false}, AgentSQLTrace:config.AgentSQLTrace{Enabled:false, Binds:false, Waits:false, TracedRuns:0}, CustomQueries:[]config.CustomQuery(nil), MetricCollectionInterval:0}
        	            	actual  : config.InstanceConfig{Server:"localhost", Port:1521, ServiceName:"XE", Username:"c##datadog", Password:"datadog", TnsAlias:"XE", TnsAdmin:"/Users/nenad.noveljic/go/src/github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle-dbm/testutil/etc/netadmin", Protocol:"", Wallet:"", DBM:true, Tags:[]string(nil), LogUnobfuscatedQueries:false, ObfuscatorOptions:obfuscate.SQLConfig{DBMS:"oracle", TableNames:true, CollectCommands:true, CollectComments:true, ReplaceDigits:false, KeepSQLAlias:false, DollarQuotedFunc:false, Cache:false}, InstantClient:false, ReportedHostname:"", QuerySamples:config.QuerySamplesConfig{Enabled:true, IncludeAllSessions:false}, QueryMetrics:config.QueryMetricsConfig{Enabled:true, CollectionInterval:60, DBRowsLimit:10000, PlanCacheRetention:15, DisableLastActive:false}, SysMetrics:config.SysMetricsConfig{Enabled:true}, Tablespaces:config.TablespacesConfig{Enabled:true}, ProcessMemory:config.ProcessMemoryConfig{Enabled:true}, SharedMemory:config.SharedMemoryConfig{Enabled:true}, ExecutionPlans:config.ExecutionPlansConfig{Enabled:false}, AgentSQLTrace:config.AgentSQLTrace{Enabled:false, Binds:false, Waits:false, TracedRuns:0}, CustomQueries:[]config.CustomQuery(nil), MetricCollectionInterval:60}
```

As I'd like to merge the PR today, I left it for now as it is, but if you think there's a better way, I'll open a separate ticket for the improvement.

Reply:
Hum yeah. Well, we could just have the expected instance config include the values that are expected to be set on `SQLConfig` but I get that you might not want/need to assert those things that aren't the immediate concern of this test. I'll leave it up to you to decide how you'd like to deal with this one.